test/etools-datepicker_test.html

```diff
@@ -20,10 +21,32 @@
 
     <script>
       suite('etools-datepicker', function() {
+        let basic
+        setup(function() {
+          basic = fixture('basic')
+        });
         test('instantiating the element works', function() {
-          var element = fixture('basic');
-          assert.equal(element.is, 'etools-datepicker');
+          assert.equal(basic.is, 'etools-datepicker');
+        });
+        test('tapping datepicker icon toggles it open', function() {
+          expect(dialog.opened).to.equal(false)
+          let calendarIcon = Polymer.dom(basic.root).querySelector('iron-icon')
+          MockInteractions.tap(calendarIcon)
+          expect(dialog.opened).to.equal(true)
         });
+        test('tapping a date selects it', function() {
+          let calendarIcon = Polymer.dom(basic.root).querySelector('iron-icon')
+          MockInteractions.tap(calendarIcon)
+          let firstDay = picker.$$('#calendar').$$('.day-item.selection')
+          let firstDayValue = firstDay.attributes.date.value
+          MockInteractions.tap(firstDay)
+          let firstDayArray = firstDayValue.split('-')
+          expect(
```

Comment:
By the time this runs, your date picker is not completely rendered. Each time you are using `MockInteractions` or you have `dom-if`/`dom-repeat` or anything else that delays the element full render you nee to use `flush` funtion to make sure the element has all data set and rendered.

```javascript
test('some test using flush', function(done) {
    // you can have some code in here fir setting up some data for your element
   flush(function() {
     // query elements, make assertions
    done(); // run done to tell wct to move to next test
   });
});
```

Reply:
Your test should be something like this to work:

```javascript
test('tapping a date selects it', function(done) {
      // set current month calendar element selector
      // calendar will always have 3 months rendered: previous, current, next
      // `picker.$$('#calendar').$$('.day-item.selection')` will return first day from previous month
      let currentDate = new Date();
      // getMonth() returns a value between 0 and 11
      let calendarCurrentMonthSelector = '.month-' + currentDate.getFullYear() + '-' + (currentDate.getMonth() + 1);

      flush(function() {
        let calendarIcon = Polymer.dom(basic.root).querySelector('iron-icon');
        // open datepicker
        MockInteractions.tap(calendarIcon);

        // select first day from current month
        let picker = basic.$$('#picker');
        let currentMonth = picker.$$('#calendar').$$(calendarCurrentMonthSelector);
        // select first day from current month (exclude disabled days in case first day is not Sunday)
        // use `querySelector`, not shadowDom here to use `$$`
        let firstDayFromCurrentMonth = currentMonth.querySelector('.month-col .day-item.selection:not([disabled])');

        // tap on first day from current month
        MockInteractions.tap(firstDayFromCurrentMonth);
        // tap ok btn to update selected day
        MockInteractions.tap(Polymer.dom(basic.root).querySelector('paper-button[dialog-confirm]'));

        // get current month first day value
        let selectedDayValue = firstDayFromCurrentMonth.attributes.date.value;
        let selectedDayArray = selectedDayValue.split('-');
        // NOTE: month in this value is from [1, 12] interval
        // we need to prepare the current month for Date.UTC format
        let currentSelectedDayMonthValueForUTCDate = selectedDayArray[1] - 1;

        // UTC is not really needed right now, you will get the same results without UTC
        expect(Date.UTC(selectedDayArray[0], currentSelectedDayMonthValueForUTCDate, selectedDayArray[2]))
            .to.equal(Date.UTC(picker.date.getFullYear(), picker.date.getMonth(), picker.date.getDate()));
        done();
      });

});
```